packages/react-components/src/organisms/DiscussionModal.tsx

```diff
@@ -53,27 +53,42 @@ const dismissButtonStyles = css({
   },
 });
 
-type QuickCheckReplyModalProps = {
+type DiscussionModalProps = {
+  title: string;
+  editorLabel: string;
+  discussionType: DiscussionType;
+  ruleMessage: string;
   onDismiss: () => void;
   discussionId: string;
-  onReplyToDiscussion: (
+  onSave: (
     id: string,
-    patch: DiscussionPatchRequest,
+    data: DiscussionPatchRequest | DiscussionCreateData,
```

Comment:
if we change this `DiscussionModalData` to be `{ text: string }`
```suggestion
    data: text,
```